src/SQL/Builder/DropSchemaObjectsSQLBuilder.php

```diff
@@ -51,4 +52,28 @@ private function buildSequenceStatements(array $sequences): array
 
         return $statements;
     }
+
+    /**
+     * @param list<string> $namespaces
+     *
+     * @return list<string>
+     */
+    private function buildNamespaceStatements(array $namespaces): array
+    {
+        if (!$this->platform->supportsSchemas()) {
+            return [];
+        }
+
+        $statements = [];
+
+        foreach ($namespaces as $namespace) {
+            if ($namespace === 'public') {
+                continue;
+            }
```

Comment:
not sure what to do with the "main" `public` schema, I don't know if this is specific to PG or not.

In PG, this schema is automatically created when the db is created, so I don't think we should drop it with `doctrine:schema:drop`

Reply:
Good question. Schemas can have different permissions in newer PostgreSQL versions and removing the defult `public` would also remove those with it, as it would with the custom schemas. I guess I'd expect it to be removed too when I want to drop schemas. :thinking:

Reply:
ok, I've removed the exception for the `public` schema

Reply:
Since I have removed the exception for `public` schema, I got an error in `PostgreSQLSchemaManagerTest::testDropWithAutoincrement()`:
```
Doctrine\DBAL\Driver\PgSQL\Exception: cannot drop schema public because other objects depend on it

in PostgreSQLSchemaManagerTest.php:364
```

And given [this comment](https://github.com/doctrine/dbal/blob/4.2.x/tests/Functional/Schema/PostgreSQLSchemaManagerTest.php#L336-L348), I think we will have problems dropping schemas which contains views, and schema "that need to be quoted".

WDYT?

Reply:
hi @SenseException @greg0ire , any chance to have some insights about this please? thanks! :pray: :pray:

Reply:
It looks like in order to correctly drop the schemas containing views, we need to implement introspection of the views in the schema.

The following is not by design, it was just never implemented.

https://github.com/doctrine/dbal/blob/c91166a91b8639a1d7bfe0f88bca0273bdbe9d2d/tests/Functional/Schema/PostgreSQLSchemaManagerTest.php#L341-L343

Reply:
https://github.com/doctrine/dbal/blob/c91166a91b8639a1d7bfe0f88bca0273bdbe9d2d/tests/Functional/Schema/PostgreSQLSchemaManagerTest.php#L339-L340

I don't remember what exactly the problem is. Quoting names that start with a digit is a workaround. I don't think it's a strict requirement if your code will be unable to drop such schemas as long as it doesn't introduce new problems. The issues with quoted names need to be fixed separately.

Reply:
yes, I was planning to fix both problems in separate PRs, thanks for your hints!

Reply:
@morozov I wanted this PR to be a fix, but implementing introspection of the views in the schema feels like a total new feature... WDYT?